docs/installation/server-config/security.rst

```diff
@@ -132,28 +133,213 @@ Configuring Authorization of KSQL with Kafka ACLs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A literal resource pattern matches resources exactly. They are case-sensitive. e.g.
+``ALLOW`` ``user Fred`` to ``READ`` the ``TOPIC`` with the ``LITERAL`` name ``users``.
+
+Here, user Fred would be allowed to read from the topic users only.
+Fred would not be allowed to read from similarly named topics such as user, users-europe, Users etc.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ACLs on Prefixed Resource Pattern
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A prefixed resource pattern matches resources where the resource name starts with the pattern's name.
+They are case-sensitive. e.g.
+``ALLOW`` ``user Bob`` to ``WRITE`` to any ``TOPIC`` whose name is ``PREFIXED`` with ``fraud-``.
+
+Here, user Bob would be allowed to write to any topic whose name starts with fraud-, e.g. fraud-us, fraud-testing and fraud-.
+Bob would not be allowed to write to topics such as production-fraud-europe, Fraud-us, etc.
+
+Required ACLs
+^^^^^^^^^^^^^
+
+The ACLs required are the same for both :ref:`Interactive and non-interactive (headless) KSQL clusters <restrict-ksql-interactive>`.
+
+KSQL always requires the following ACLs for its internal operations and data management:
+
+- The ``DESCRIBE_CONFIGS`` operation on the ``CLUSTER`` resource type.
+- The ``ALL`` operation on all internal ``TOPIC``s ``PREFIXED`` with ``_confluent-ksql-<ksql.service.id>``.
+- The ``ALL`` operation on all internal ``GROUP``s ``PREFIXED`` with ``_confluent-ksql-<ksql.service.id>``.
+
+Where ``ksql.service.id`` can be configured in the KSQL configuration and defaults to ``default_``.
+
+In addition to the general permissions above, KSQL also needs permissions to perform the actual processing of your data.
+Here, KSQL needs permissions to read data from your desired input topics and/or permissions to write data to your desired output topics:
+
+- The ``READ`` operation on any input topics.
+- The ``WRITE`` operation on any output topics.
+- The ``CREATE`` operation on any output topics that do not already exist.
+
+Often output topics from one query form the inputs to others. KSQL will require ``READ`` and ``WRITE`` permissions for such topics.
+
+The set of input and output topics that a KSQL cluster requires access to will depend on your use case and
+whether the KSQL cluster is configured in
+:ref:`interactive <config-security-ksql-acl-interactive_post_ak_2_0>` or :ref:`non-interactive <config-security-ksql-acl-headless_post_ak_2_0>` mode.
+
+.. _config-security-ksql-acl-headless_post_ak_2_0:
+
+Non-Interactive (headless) KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:ref:`Non-interactive KSQL clusters <restrict-ksql-interactive>` run a known set of SQL statements, meaning the set
+of input and output topics is well defined. Add the ACLs required to allow KSQL access to these topics.
+
+For example, given the following setup:
+
+- A 3-node KSQL cluster
+- Running on hosts with IPs 198.51.100.0, 198.51.100.1, 198.51.100.2
```

Comment:
I'd collapse these two bullets and reword to:

- A 3-node KSQL cluster with KSQL servers running on IPs 198.51.100.0, 198.51.100.1, 198.51.100.2